Tests/ApolloSQLiteTests/CachePersistenceTests.swift

```diff
@@ -63,70 +63,105 @@ class CachePersistenceTests: XCTestCase {
   func testPassInConnectionDoesNotThrow() {
     XCTAssertNoThrow(try SQLiteNormalizedCache(db: Connection()))
   }
+}
 
-  func testClearCache() {
-    let query = HeroNameQuery()
+// MARK: Cache clearing
+
+extension CachePersistenceTests {
+  func testClearMatchingKeyPattern() {
+    let notFoundExpectation = self.expectation(description: "records should not exist")
+
+    self.testCacheClearing(withPolicy: .allMatchingKeyPattern("*hero*")) { client, _ in
+      client.store.withinReadTransaction {
+        $0.loadRecords(forKeys: ["QUERY_ROOT.hero", "QUERY_ROOT.hero(episode:EMPIRE)"]) { result in
+          defer { notFoundExpectation.fulfill() }
+
+          do {
+            let results = try result.get()
+            XCTAssertTrue(results.allSatisfy({ $0 == nil }))
```

Comment:
why should these ones be `nil` instead of getting a `.missingValue` error?

Reply:
I wasn't getting `.missingValue` I was getting `[nil, nil]`, and I was coding at 02:00 my time so admittedly this was a bit rushed and not written to a high quality 😅

Reply:
so that indicates that the values stored for each key are `nil`, rather than the whole key and value being removed, which is what I believe should happen. Otherwise you could potentially get a `nil` value for something that's non-nullable when you're trying to pull it out of the cache.

Reply:
The difference in the two unit tests is that one is going through the whole GraphQL executor by calling `fetch` on the client, while the other (the one this comment was left on) was accessing the cache direction in a read transaction.

I've re-written this to be more consistent and more direct at what's being looked for.